### PR TITLE
Unitest2pytest entities file info

### PR DESCRIPTION
**Pull request by wkrt7**
Signed-off-by: Wiktor Wasowicz 799637+wkrt7@users.noreply.github.com 
## Related Issues/PRs

 #7201 

## What changes are proposed in this pull request?

Converted entities module's tests to pytest

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/entities/conftest.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+@pytest.fixture(scope="module")
+def params():
+    yield [Param(random_str(10), random_str(random_int(10, 35))) for _ in range(10)]
+
+
+@pytest.fixture(scope="module")
+def tags():
+    yield [RunTag(random_str(10), random_str(random_int(10, 35))) for _ in range(10)]
```
Comment (harupy) [inline tests/entities/conftest.py:31]: I don't think we need to fixturize everything. If you don't need teardown, a plain function should normally suffice.

File: tests/entities/test_experiment.py [hunk 2]
```diff
@@ -1,67 +1,91 @@
...
+
+
+@pytest.fixture
+def creation_time(scope="module"):
+    yield get_current_time_millis()
+
+
+@pytest.fixture
+def last_update_time(scope="module"):
+    yield get_current_time_millis()
```
Comment (harupy) [inline tests/entities/test_experiment.py:44]: Can we directly use `get_current_time_millis`?

File: tests/entities/test_experiment.py [hunk 3]
```diff
@@ -1,67 +1,91 @@
...
+    yield LifecycleStage.ACTIVE
+
+
+@pytest.fixture
+def location(scope="module"):
+    yield random_file(".json")
+
+
+@pytest.fixture
+def creation_time(scope="module"):
```
Comment (harupy) [inline tests/entities/test_experiment.py:38]: Can we directly use get_current_time_millis?

File: tests/entities/test_param.py [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
-        assert param.value == value
+def _check(param, key, value):
+    assert isinstance(param, Param)
+    assert param.key == key
+    assert param.value == value
+
+
+@pytest.fixture(scope="module")
+def key():
+    yield random_str(random_int(10, 25))  # random string on size in range [10, 25]
```
Comment (harupy) [inline tests/entities/test_param.py:14]: I think we can directly use `random_str(random_int(10, 25))`.

File: tests/entities/test_experiment.py (hunk 2 above)
Comment (wkrt7) [reply tests/entities/test_experiment.py]: solved in https://github.com/wkrt7/mlflow/commit/3a76eb2e91490a13ea70ca338c5a84dd491f8c26

File: tests/entities/test_experiment.py (hunk 3 above)
Comment (wkrt7) [reply tests/entities/test_experiment.py]: solved in https://github.com/wkrt7/mlflow/commit/3a76eb2e91490a13ea70ca338c5a84dd491f8c26

File: tests/entities/test_param.py (hunk 4 above)
Comment (wkrt7) [reply tests/entities/test_param.py:14]: solved in https://github.com/wkrt7/mlflow/commit/3a76eb2e91490a13ea70ca338c5a84dd491f8c26

Comment (harupy) [reply tests/entities/test_param.py:14]: @wkrt7 Can we also remove `key` and `value`?

File: tests/entities/conftest.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+from tests.helper_functions import random_str, random_int
+
+from mlflow.entities import RunInfo, LifecycleStage, RunStatus
+from mlflow.entities import Metric, RunData, Param, RunTag
+from mlflow.utils.time_utils import get_current_time_millis
+
+
+@pytest.fixture(scope="module")
+def test_run_data():
```
Comment (harupy) [inline tests/entities/conftest.py:13]:
```suggestion
def run_data():
```

File: tests/entities/conftest.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            timestamp=get_current_time_millis() + random_int(-1e4, 1e4),
+            step=random_int(),
+        )
+    ]
+    params = [Param(random_str(10), random_str(random_int(10, 35))) for _ in range(10)]
+    tags = [RunTag(random_str(10), random_str(random_int(10, 35))) for _ in range(10)]
+
+    rd = RunData(metrics=metrics, params=params, tags=tags)
+
+    yield rd, metrics, params, tags
```
Comment (harupy) [inline tests/entities/conftest.py:27]:
```suggestion
    return rd, metrics, params, tags
```

File: tests/entities/conftest.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+
+@pytest.fixture(scope="module")
+def test_run_info():
```
Comment (harupy) [inline tests/entities/conftest.py:31]:
```suggestion
def run_info():
```

File: tests/entities/conftest.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        run_name=run_name,
+        experiment_id=experiment_id,
+        user_id=user_id,
+        status=status,
+        start_time=start_time,
+        end_time=end_time,
+        lifecycle_stage=lifecycle_stage,
+        artifact_uri=artifact_uri,
+    )
+    yield (
```
Comment (harupy) [inline tests/entities/conftest.py:53]:
```suggestion
    return (
```

File: tests/entities/conftest.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import uuid
+
+from tests.helper_functions import random_str, random_int
+
+from mlflow.entities import RunInfo, LifecycleStage, RunStatus
+from mlflow.entities import Metric, RunData, Param, RunTag
+from mlflow.utils.time_utils import get_current_time_millis
+
+
+@pytest.fixture(scope="module")
```
Comment (harupy) [inline tests/entities/conftest.py:12]: Does this fixture really have to be module-scoped?

File: tests/entities/test_param.py (hunk 4 above)
Comment (wkrt7) [reply tests/entities/test_param.py:14]: I moved key and value to the test function

File: tests/entities/conftest.py (hunk 7 above)
Comment (wkrt7) [reply tests/entities/conftest.py:31]: Changed the fixture's name

File: tests/entities/conftest.py (hunk 9 above)
Comment (wkrt7) [reply tests/entities/conftest.py:12]: No, changed it to function.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**